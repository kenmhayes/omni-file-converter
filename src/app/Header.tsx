import React from 'react';
import { Nav, Navbar } from 'react-bootstrap';
import omniLogo from '../assets/svg/omni-logo.svg';

/**
 * An element providing navigation bar features, to be displayed at the top of a webpage
 * @returns
 *   a react element
 */
function Header() {
  return (
    <Navbar bg="dark" data-bs-theme="dark" className="shadow-sm mb-3">
      <div className="container">
        <Navbar.Brand href="/" className="d-flex align-items-center">
          <svg width="32" height="32" fill="currentColor">
            <use xlinkHref={`${omniLogo}#omni-logo`} />
          </svg>
          <span>Omni Converter</span>
        </Navbar.Brand>
        <Nav>
          <Nav.Link href="/" className="btn btn-outline-light">
            Convert
          </Nav.Link>
          <Nav.Link href="myfiles">
            My Files
          </Nav.Link>
        </Nav>
      </div>
    </Navbar>
  );
}

export default Header;
