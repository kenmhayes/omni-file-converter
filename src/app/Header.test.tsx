import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Header from './Header';

it('renders a navbar with links', () => {
  // A router is necessary for Links to work
  const component = create(
    <MemoryRouter>
      <Header/ >
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
