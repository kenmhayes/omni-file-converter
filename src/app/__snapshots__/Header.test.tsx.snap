// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a navbar with links 1`] = `
<nav
  className="shadow-sm mb-3 navbar navbar-expand navbar-light bg-dark"
  data-bs-theme="dark"
>
  <div
    className="container"
  >
    <a
      className="d-flex align-items-center navbar-brand"
      href="/"
    >
      <svg
        fill="currentColor"
        height="32"
        width="32"
      >
        <use
          xlinkHref="omni-logo.svg#omni-logo"
        />
      </svg>
      <span>
        Omni Converter
      </span>
    </a>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="btn btn-outline-light nav-link"
        data-rr-ui-event-key="/"
        disabled={false}
        href="/"
        onClick={[Function]}
      >
        Convert
      </a>
      <a
        className="nav-link"
        data-rr-ui-event-key="myfiles"
        disabled={false}
        href="myfiles"
        onClick={[Function]}
      >
        My Files
      </a>
    </div>
  </div>
</nav>
`;
